src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java

```diff
@@ -710,93 +794,98 @@ private void deleteChildEntities(Table childTable, Field keyField, String keyVal
-        for (String tripId : timesForTripIds.keySet()) {
-            // Initialize travel time with previous stop time value.
-            int cumulativeTravelTime = timesForTripIds.get(tripId);
-            for (PatternStop patternStop : patternStops) {
-                // Gather travel/dwell time for pattern stop (being sure to check for missing values).
-                int travelTime = patternStop.default_travel_time == Entity.INT_MISSING ? 0 : patternStop.default_travel_time;
-                int dwellTime = patternStop.default_dwell_time == Entity.INT_MISSING ? 0 : patternStop.default_dwell_time;
-                int oneBasedIndex = 1;
-                // Increase travel time by current pattern stop's travel and dwell times (and set values for update).
-                cumulativeTravelTime += travelTime;
-                updateStopTimeStatement.setInt(oneBasedIndex++, cumulativeTravelTime);
-                cumulativeTravelTime += dwellTime;
-                updateStopTimeStatement.setInt(oneBasedIndex++, cumulativeTravelTime);
-                updateStopTimeStatement.setString(oneBasedIndex++, tripId);
-                updateStopTimeStatement.setInt(oneBasedIndex++, patternStop.stop_sequence);
-                stopTimesTracker.addBatch();
-            }
+        // Prepare the statement and set statement parameters
+        PreparedStatement statement = connection.prepareStatement(sql);
+        int oneBasedIndex = 1;
+        int travelTime = patternLocation.flex_default_travel_time;
+        int timeInLocation = patternLocation.flex_default_zone_time;
+        int arrivalTime = previousTravelTime + travelTime;
+        // Setting start of the pickup window (when the vehicle arrives in the flex location)
+        statement.setInt(oneBasedIndex++, arrivalTime);
+        // Set the end of the pickup window (when the vehicle leaves the flex location)
+        statement.setInt(oneBasedIndex++, arrivalTime + timeInLocation);
+
+        // Set trip id either from params or all
+        if (tripId != null) {
+            statement.setString(oneBasedIndex++, tripId);
+        } else {
+            statement.setString(oneBasedIndex++, "t.trip_id");
         }
-        return stopTimesTracker.executeRemaining();
+        // Set "where clause" with value for pattern_id and stop_sequence
+        statement.setString(oneBasedIndex++, patternLocation.pattern_id);
+        // In the editor, we can depend on stop_times#stop_sequence matching pattern_stops#stop_sequence because we
+        // normalize stop sequence values for stop times during snapshotting for the editor.
+        statement.setInt(oneBasedIndex++, patternLocation.stop_sequence);
+        // Log query, execute statement, and log result.
+        LOG.debug(statement.toString());
+        int entitiesUpdated = statement.executeUpdate();
+        LOG.debug("{} stop_time arrivals/departures updated", entitiesUpdated);
```

Comment:
I believe we're expecting potential changes here as we work through our bug backlog, but will leave this open in case we don't touch this section again

Reply:
I have extracted.